package/src/main.rs

```diff
@@ -403,6 +405,56 @@ impl BuildContext {
             cargo_output_bin_dir: output_bin_dir,
         })
     }
+
+    fn ensure_target(&mut self) -> ExitResult {
+        if !self.target_prepared {
+            build_step!(
+                "Ensuring --target {target} is available",
+                target = self.target
+            );
+            execute(Command::new("rustup").args(["target", "add", &self.target]))?;
+            self.target_prepared = true;
+        }
+        Ok(())
+    }
+
+    fn obtain_ptex(&mut self, dest_dir: &Path) -> Result<PathBuf, Exit> {
+        if self.ptex_repo.is_some() {
```

Comment:
I was not clever enough to be able use an `if let Some(...) = ...` binding here and satisfy the borrow checker; so this safe, but unsatisfying is_some / unwrap unconditionally a few lines lower combo.

Reply:
Got smarter, Cell allows for a partially mutable struct. Fixed.